pandas/_libs/window.pyx

```diff
@@ -1391,7 +1392,19 @@ def roll_quantile(ndarray[float64_t, cast=True] input, int64_t win,
 
         if nobs >= minp:
             idx = int(quantile * <double>(nobs - 1))
-            output[i] = skiplist.get(idx)
+
+            # Exactly last point
+            if idx == nobs - 1:
+                output[i] = skiplist.get(idx)
+
+            # Interpolated quantile
+            else:
+                qlow = (<double> idx) / (<double>(nobs - 1))
+                qhigh = (<double> (idx + 1)) / (<double>(nobs - 1))
+                vlow = skiplist.get(idx)
+                vhigh = skiplist.get(idx + 1)
+                output[i] = vlow + (vhigh - vlow) * \
```

Comment:
The computation time is probably dominated by the skip list, but it still feels wrong to have three divisions in the innermost loop. And I think we can actually get rid of all of them! If this is correct:

```
qlow = i / (nobs - 1)
qhigh = (i + 1) / (nobs - 1)
(quantile - qlow) / (qhigh - qlow) = (q - i / (nobs - 1)) / ((i + 1 - i) / (nobs - 1)) = (nobs - 1) * q - i
```

It should be possible to simply write

```
output[i] = vlow + (vhigh - vlow) * (quantile * (nobs - 1) - idx)
```

It should be faster, although probably not much, but on top of that I think that the fact that `quantile * (nobs - 1) - idx` is the fractional part of `quantile * (nobs - 1)` makes the code a little more self explanatory?